engines/sword1/screen.h

```diff
@@ -76,30 +76,53 @@ class Screen {
 	void clearScreen();
 	void useTextManager(Text *pTextMan);
 	void draw();
+	void initFadePaletteServer();
 
 	void quitScreen();
 	void newScreen(uint32 screen);
 
 	void setScrolling(int16 offsetX, int16 offsetY);
 	void addToGraphicList(uint8 listId, uint32 objId);
 
-	void fadeDownPalette();
-	void fadeUpPalette();
-	void fnSetPalette(uint8 start, uint16 length, uint32 id, bool fadeUp);
-	bool stillFading();
-	void fullRefresh();
+	void startFadePaletteDown(int speed);
+	void startFadePaletteUp(int speed);
+	void fadePalette();
+	void fnSetPalette(uint8 start, uint16 length, uint32 id);
+	void fnSetFadeTargetPalette(uint8 start, uint16 length, uint32 id, bool toBlack = false);
+	int16 stillFading();
+	void fullRefresh(bool soft = false);
 
 	bool showScrollFrame();
 	void updateScreen();
 	void showFrame(uint16 x, uint16 y, uint32 resId, uint32 frameNo, const byte *fadeMask = NULL, int8 fadeStatus = 0);
 
 	void fnSetParallax(uint32 screen, uint32 resId);
 	void fnFlash(uint8 color);
-	void fnBorder(uint8 color);
 
 	static void decompressHIF(uint8 *src, uint8 *dest);
 
 private:
+	// The original values are 6-bit RGB numbers, so they have to be shifted
+	const byte _white[3]  = { 63 << 2, 63 << 2, 63 << 2};
+	const byte _red[3]    = { 63 << 2, 0  << 2, 0  << 2};
+	const byte _blue[3]   = { 0  << 2, 0  << 2, 63 << 2};
+	const byte _yellow[3] = { 63 << 2, 63 << 2, 0  << 2};
+	const byte _green[3]  = { 0  << 2, 63 << 2, 0  << 2};
+	const byte _purple[3] = { 32 << 2, 0  << 2, 32 << 2};
```

Comment:
32 or 31? E.g. one bit set or 5, following 63

Reply:
Just re-checked the source code for good measure: it actually uses 32 (and 63 above) 🙂